### ISSUE TITLE
Clarification needed in "Network Access"

### ISSUE THREAD

**Issue by zoe-translates** [common-workflow-language/user_guide#396]
https://github.com/common-workflow-language/user_guide/blob/7b71012f6d286fbea44e052a54a1d5977eb9f3ca/src/topics/command-line-tool.md?plain=1#LL70C1-L71C40

> If a command-line tool is written **manually** in CWL v1.1+, there is a need to specify when network access is required.

Here is it actually relevant whether "a command-line tool... in CWL v1.1+" is written "manually" (as opposed to output by code-generation)?

Based on what I know (not much), this may mean to say

"For a command-line tool written *de novo* following the CWL spec v1.1 or above, it is necessary to explicitly specify the network access requirement when applicable. For CWL tools originally targeting v1.0, when 'upgraded' (by a code-generation tool? please illuminate), network access requirement is automatically added to them“,

or 

"... when 'upgraded' (i.e. when a CWL file specifying `1.0` as version is being executed by a higher-versioned cwl runner), network access requirement is implicitly added at runtime, no matter whether actual access is going to be made by the tool."

Is one of my understandings accurate?

Asking because it's difficult to translate correctly without further clarity.

**Comment by mr-c** [comment]
> "For a command-line tool written _de novo_ following the CWL spec v1.1 or above, it is necessary to explicitly specify the network access requirement when applicable. For CWL tools originally targeting v1.0, when 'upgraded' (by a code-generation tool? please illuminate), network access requirement is automatically added to them“,

This is correct. `cwltool` and engines that use it can auto-upgrade. There is also a standalone upgrader that also adds the network access requirement automatically. 

<https://pypi.org/project/cwl-upgrader/>

Can you open a pull request to improve the English wording?

(Many thanks!)
